docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/upgrade-guide-from-5.4-to-6.0-generic.rst

```diff
@@ -0,0 +1,359 @@
+
+.. |SCYLLA_DEB_NEW_REPO| replace:: ScyllaDB deb repo (|DEBIAN_NEW_REPO|_, |UBUNTU_NEW_REPO|_)
+
+.. |SCYLLA_RPM_NEW_REPO| replace:: ScyllaDB rpm repo
+.. _SCYLLA_RPM_NEW_REPO: https://www.scylladb.com/download/?platform=centos&version=scylla-6.0
+
+.. |ROLLBACK| replace:: rollback
+.. _ROLLBACK: ./#rollback-procedure
+
+.. |SCYLLA_METRICS| replace:: Scylla Metrics Update - Scylla 5.4 to 6.0
+.. _SCYLLA_METRICS: ../metric-update-5.4-to-6.0
+
+=============================================================================
+Upgrade |SCYLLA_NAME| from |SRC_VERSION| to |NEW_VERSION|
+=============================================================================
+
+This document is a step-by-step procedure for upgrading from |SCYLLA_NAME| |SRC_VERSION| 
+to |SCYLLA_NAME| |NEW_VERSION|, and rollback to version |SRC_VERSION| if required.
+
+This guide covers upgrading ScyllaDB on Red Hat Enterprise Linux (RHEL), CentOS, Debian, 
+and Ubuntu. See :doc:`OS Support by Platform and Version </getting-started/os-support>` 
+for information about supported versions.
+
+It also applies when using ScyllaDB official image on EC2, GCP, or Azure.
+
+Before You Upgrade ScyllaDB
+==============================
+
+**Upgrade Your Driver**
+
+If you're using a :doc:`ScyllaDB driver </using-scylla/drivers/cql-drivers/index>`, 
+upgrade the driver before you upgrade ScyllaDB. The latest two versions of each driver 
+are supported.
+
+**Upgrade ScyllaDB Monitoring Stack**
+
+If you're using the ScyllaDB Monitoring Stack, verify that your Monitoring Stack 
+version supports the ScyllaDB version to which you want to upgrade. See 
+`ScyllaDB Monitoring Stack Support Matrix <https://monitoring.docs.scylladb.com/stable/reference/matrix.html>`_.
+  
+We recommend upgrading the Monitoring Stack to the latest version.
+
+**Check Feature Updates**
+
+See the ScyllaDB Release Notes for the latest updates. The Release Notes are published 
+at the `ScyllaDB Community Forum <https://forum.scylladb.com/>`_.
+
+
+Upgrade Procedure
+=================
+
+A ScyllaDB upgrade is a rolling procedure that does **not** require full cluster shutdown.
+For each of the nodes in the cluster, serially (i.e., one node at a time), you will:
+
+#. Check that the cluster's schema is synchronized.
+#. Drain the node and backup the data.
+#. Backup the configuration file.
+#. Stop ScyllaDB.
+#. Download and install new ScyllaDB packages.
+#. Start ScyllaDB.
+#. Validate that the upgrade was successful.
+
+Apply the procedure **serially** on each node. Do not move to the next 
+node before validating that the node you upgraded is up and running the new version.
+
+**During** the rolling upgrade, it is highly recommended:
+
+* Not to use the new |NEW_VERSION| features.
+* Not to run administration functions, such as repairs, refresh, rebuild, or add 
+  or remove nodes.
+* Not to apply schema changes.
+
+**After** the upgrade, enable consistent topology updates. 
```

Comment:
Sorry for not noticing this before -

I'm actually missing the "Validate Raft setup" section which was present in the "upgrade guide to 5.5". This one:
```rst
After Upgrading Every Node
===============================

Skip this step if you upgraded from 5.2 to 5.4 with default settings

This section only applies if you manually disabled the ``consistent_cluster_management`` 
option before upgrading from version 5.2. to 5.4.

.. _validate-raft-setup-enabled-default:

Validate Raft Setup
-------------------------

Enabling Raft causes the ScyllaDB cluster to start an internal Raft 
initialization procedure as soon as every node is upgraded to the new version. 
The goal of that procedure is to initialize data structures used by the Raft 
algorithm to consistently manage cluster-wide metadata, such as table schemas.

Assuming you performed the rolling upgrade procedure correctly (in particular, 
ensuring that the schema is synchronized on every step), and if there are no 
problems with cluster connectivity, that internal procedure should take a few 
seconds to finish. However, the procedure requires full cluster availability.
If one of the nodes fails before the procedure finishes (for example, due to 
a hardware problem), the process may get stuck, which may prevent schema or 
topology changes in your cluster.

Therefore, following the rolling upgrade, you must verify that the internal 
Raft initialization procedure has finished successfully by checking the logs 
of every ScyllaDB node. If the process gets stuck, manual intervention is 
required.

Refer to the 
:ref:`Verifying that the internal Raft upgrade procedure finished successfully <verify-raft-procedure>` 
section for instructions on verifying that the procedure was successful and 
proceeding if it gets stuck.
```

This is still relevant, if someone prevented Raft from getting enabled in 5.4.

And it must be done before one attempts to enable consistent topology updates.

---


In fact this leads me to a question: why don't we simply rename the existing upgrade-to-5.5 guide, and add things to it as needed? The upgrade-to-6.0 guide should basically be the same as the upgrade-to-5.5 guide, with some minor changes and additions such as replacing the string "5.5" with "6.0".

With this approach of "add new guide, remove old one" it's easy to miss important parts -- like the validate section that I mentioned above.

Reply:
> I did it this way just because I thought we should see it in the commit log (there's information about 5.5 guides being added; wanted to indicate they were removed, as they should be).

Renaming 5.5 to 6.0 would be just as clear.

> Anyway, it wouldn't solve the issue, because I removed the "Validate Raft Setup" section on purpose.

It may not change much from your perspective, but it changes a lot from reviewers' perspective.

With this "nuke one article, add another" approach, it's hard to see the difference between the old guide (upgrade-to-5.5) and new guide (upgrade-to-6.0). With "modify existing article" approach I would clearly see the diff and catch immediately the fact that you removed the section.